src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -1046,15 +1054,85 @@ public Optional<File> findRapidScanConfig() throws OperationException {
                 .findRapidScanConfig(directoryManager.getSourceDirectory())
         );
     }
-    
-    public BomStatusScanView waitForBomScanCompletion(BlackDuckRunData blackDuckRunData, HttpUrl scanUrl) throws OperationException {
+
+    public File getScanCliOutputLogFile() {
+        File blackDuckScanOutputDirectory = this.fileFinder.findFile(directoryManager.getScanOutputDirectory(), "BlackDuckScanOutput");
+        if (blackDuckScanOutputDirectory != null) {
+            File scanIdDirectory = this.fileFinder.findFile(blackDuckScanOutputDirectory, "*");
+            if (scanIdDirectory != null) {
+                File scanLogDirectory = this.fileFinder.findFile(scanIdDirectory, "log");
+                if (scanLogDirectory != null) {
+                    return this.fileFinder.findFile(scanLogDirectory, "*.log");
+                }
+            }
+        }
+        return null;
+    }
+
+    public int countSignatureScannerBdioChunks() {
+        try {
+            File scanCliOutputLogFile = getScanCliOutputLogFile();
+            if (scanCliOutputLogFile == null) {
+                return 0;
+            }
+            BufferedReader reader = new BufferedReader(new FileReader(scanCliOutputLogFile));
+
+            Pattern pattern = Pattern.compile("scanNodeList\\.size\\(\\)=(\\d+).*scanLeafList\\.size\\(\\)=(\\d+)");
+            long scanNodeCount = 0;
+            long scanLeafCount = 0;
+
+            String line = reader.readLine();
+            while (line != null) {
+                Matcher matcher = pattern.matcher(line);
```

Comment:
I've made these methods as private (after our discussion on Slack). 
Regarding the case that if scan CLI's log output file structure changes, the regex pattern simply won't match (`mathcer.find()` would be false) and `sumOfScanNodesAndLeaves` would be 0 (since node count and leaf count are initialized as 0). On any other IOException, 0 is returned.